lib/OpenQA/WebAPI/Auth/OpenID.pm

```diff
@@ -105,9 +105,16 @@ sub auth_response {
 
     my $msg = "The openID server doesn't respond";
 
+    my $err_handler = sub {
+        my ($err, $txt) = @_;
+        $self->app->log->error($err, $txt);
+        $self->flash(error => "$err: $txt");
+        return (error => 0);
+    };
+
     $csr->handle_server_response(
         not_openid => sub {
-            die "Not an OpenID message";
+            return $err_handler->("Failed to login", "OpenID provider returned invalid data. Please retry again later");
```

Comment:
Why should we retry later? I think the problem is not a temporary one but a broken openID server. Sure, the problem is most likely not within openQA. Can we give more information to the user?